### PR TITLE
Fix issue #24

### PR DESCRIPTION
**Pull request by eleidan**
@alazarchuk , @epmlys , @rubycop , @afurm

### CODE REVIEW

File: app/assets/javascripts/views/vacation_request_form.js [hunk 1]
```diff
@@ -53,6 +53,43 @@ App.Views.VacationRequestForm = Backbone.View.extend({
     this.updateFormState();
   },
 
+  onError: function(model, response, options) {
+    // TODO: show error messages to user
```
Comment (alazarchuk) [inline app/assets/javascripts/views/vacation_request_form.js]: Add console.error to output response

File: app/assets/javascripts/views/vacation_request_form.js [hunk 2]
```diff
@@ -53,6 +53,43 @@ App.Views.VacationRequestForm = Backbone.View.extend({
...
+    // TODO: add some tests
+    // Clear model to set it as a new one,
+    // and initialize it with form data.
+    // Otherwise, the model is initialized with response data and a next save()
+    // will emit PUT request (update) instead of POST (create)
+    this.clearModel();
+    this.fetchFormData();
+  },
+
+  fetchFormData: function () {
```
Comment (alazarchuk) [inline app/assets/javascripts/views/vacation_request_form.js]: Any kind of validation, at least value is not empty?

Comment (eleidan) [inline app/assets/javascripts/views/vacation_request_form.js]: The validation is triggered by `this.model.save()` method, this is the default behaviour of BB models. And the `App.Models.VacationRequest` implements `validation()` method, so the validation is provided.
There an issue #35 that should be resolved later. And during solution implementation for that issue all the validation related errors will be shown to users.

To solve this problem, the following lines should be added to the code:
```JS
this.listenTo(this.model, 'invalid', this.onInvalid);
```
```JS
onInvalid: function(model, response, options) {
   console.log(model.validationError);
}  
```

File: app/assets/javascripts/views/vacation_request_form.js (hunk 1 above)
Comment (eleidan) [inline app/assets/javascripts/views/vacation_request_form.js]: Done.

**Pull request merged**